app/javascript/mastodon/components/featured_carousel.tsx

```diff
@@ -0,0 +1,182 @@
+        ImmutableList(),
+      ) as ImmutableList<string>,
+  );
+
+  // Handle slide change
+  const [slideIndex, setSlideIndex] = useState(0);
+  const handleSlideChange = useCallback(
+    (direction: number) => {
+      const max = pinnedStatuses.size - 1;
+      setSlideIndex((prev) => {
+        const newIndex = prev + direction;
+        if (newIndex < 0) {
+          return max;
+        } else if (newIndex > max) {
+          return 0;
+        }
+        return newIndex;
+      });
+    },
+    [pinnedStatuses.size],
+  );
+
+  // Handle swiping animations
+  const { x } = useSpring({ x: `-${slideIndex * 100}%` });
+  const bind = useDrag(({ swipe: [swipeX] }) => {
+    handleSlideChange(swipeX * -1); // Invert swipe as swiping left loads the next slide.
+  });
+  const handlePrev = useCallback(() => {
+    handleSlideChange(-1);
+  }, [handleSlideChange]);
+  const handleNext = useCallback(() => {
+    handleSlideChange(1);
+  }, [handleSlideChange]);
+
+  // Handle wrapper height animation
+  const [wrapperStyles, wrapperSpringApi] = useSpring(() => ({
+    height: 100,
+  }));
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const wrapperEle = wrapperRef.current;
+  useEffect(() => {
+    if (!wrapperEle) {
+      return;
+    }
+    const currentSlideEle = wrapperEle.querySelector(
+      `[data-index="${slideIndex}"] > div`,
+    );
+
+    if (!currentSlideEle) {
+      return;
+    }
+    const height = currentSlideEle.scrollHeight;
+    if (!height) {
+      return;
+    }
+
+    void wrapperSpringApi.start({ height });
+  }, [slideIndex, wrapperSpringApi, wrapperEle]);
+
+  if (!accountId || pinnedStatuses.isEmpty()) {
+    return null;
+  }
+
+  return (
+    <div
+      className='featured-carousel'
+      {...bind()}
+      aria-roledescription='carousel'
+      aria-labelledby='featured-carousel-title'
+      role='region'
+    >
+      <div className='featured-carousel__header'>
+        <h4 className='featured-carousel__title' id='featured-carousel-title'>
+          {pinnedStatuses.size > 1 ? (
+            <FormattedMessage
+              id='featured_carousel.header.plural'
+              defaultMessage='Pinned Posts'
+            />
+          ) : (
+            <FormattedMessage
+              id='featured_carousel.header.single'
+              defaultMessage='Pinned Post'
+            />
+          )}
+        </h4>
+        {pinnedStatuses.size > 1 && (
+          <>
+            <IconButton
+              title={intl.formatMessage(messages.previous)}
+              icon='chevron-left'
+              iconComponent={ChevronLeftIcon}
+              onClick={handlePrev}
+            />
+            <FormattedMessage id='featured_carousel.post' defaultMessage='Post'>
+              {(text) => <span className='sr-only'>{text}</span>}
+            </FormattedMessage>
+            <span aria-live='polite'>
+              {slideIndex + 1} / {pinnedStatuses.size}
+            </span>
```

Comment:
Include the "Post" in the live region for it to be announced alongside the numbers:
```suggestion
            <span aria-live='polite'>
              <FormattedMessage id='featured_carousel.post' defaultMessage='Post'>
                {(text) => <span className='sr-only'>{text}</span>}
              </FormattedMessage>
              {slideIndex + 1} / {pinnedStatuses.size}
            </span>
```